README.md

```diff
@@ -0,0 +1,103 @@
+# TreeTables plug-in for jQuery
+TreeTables is a jQuery plugin that enhances the functionality of the
+popular [DataTables](https://github.com/DataTables/DataTables) plugin.
+
+DataTables does not support tree data by default, so this plugin adds
+that support.
+
+## Usage
+
+Include the following scripts on your page:
+
+```
+<script src="node_modules/jquery/dist/jquery.js"></script>
+<script src="node_modules/datatables.net/js/jquery.dataTables.js"></script>
+<script src="treeTable.js"></script>
+```
+
+And the following css in your document head:
+```
+    <link rel="stylesheet" href="node_modules/datatables.net-dt/css/jquery.dataTables.css"/>
+    <link rel="stylesheet" href="tree-table.css"/>
+```
+
+### Basic Usage
+
+```
+ const organisationData = [
+            {key: 1, parent: 0, name: "CEO", hasChild: true},
+            {key: 2, parent: 1, name: "CTO", hasChild: true},
+            {key: 3, parent: 2, name: "developer", hasChild: false},
+            {key: 4, parent: 1, name: "CFO", hasChild: false}
+        ];
+
+        $('#my-table').treeTable({
+            "data": myData,
+            "columns": [
+                {
+                    "data": "name"
+                }
+            ]
+        });
+```
+
+Data provided to the table must include the following fields:
+* key: number - a unique row identifier
+* parent: number - the key of this row's parent row
+* hasChild: bool - does this row have any children
+
+### Options
+TreeTable options are all DataTable options plus:
+* collapsed: bool - whether to start with all child rows collapsed
+
+```
+        $('#my-table').treeTable({
+            "data": myData,
+            "collapsed": true,
+            "columns": [
+                {
+                    "data": "name"
+                }
+            ]
+        });
+```
+
+Please note that the TreeTable plugin adds 4 invisible columns to the table.
```

Comment:
Omg of course. This is a great idea. The toggle column has to go first because visually it is first, but that just means the user cols can be 1-indexed.